tests/decomposition/test_transforms_decompose.py

```diff
@@ -0,0 +1,308 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for the new experimental graph-based decomposition system integrated with `qml.transforms.decompose`."""
```

Comment:
```suggestion
"""Tests for the new experimental graph-based decomposition system integrated with `qml.transforms.decompose`."""
```
I think (almost) all of them are System Tests instead of Unit Tests here right? I'm wondering if it is even possible to have unit tests for decomposistion module here, in the sense that no qml device is used but only involving whatever is needed by new methods

Reply:
You're completely right! These are all system tests :D